cmd/podman/images/save.go

```diff
@@ -91,11 +96,33 @@ func save(cmd *cobra.Command, args []string) error {
 		return errors.Errorf("--compress can only be set when --format is either 'oci-dir' or 'docker-dir'")
 	}
 	if len(saveOpts.Output) == 0 {
-		fi := os.Stdout
-		if terminal.IsTerminal(int(fi.Fd())) {
-			return errors.Errorf("refusing to save to terminal. Use -o flag or redirect")
+		pipeFile := "/tmp/saveio"
+		saveOpts.Output = pipeFile
+		err := os.Remove(pipeFile)
+		if err != nil && !os.IsNotExist(err) {
+			return errors.Wrapf(err, "error removing named pipe")
 		}
-		saveOpts.Output = "/dev/stdout"
+		err = syscall.Mkfifo(pipeFile, 0666)
+		if err != nil {
+			return errors.Wrapf(err, "error creating named pipe")
+		}
+		defer os.Remove(pipeFile)
+		wg := sync.WaitGroup{}
+		wg.Add(1)
+		go func() {
+			fpipe, _ := os.OpenFile(pipeFile, os.O_RDWR, os.ModeNamedPipe)
+			defer fpipe.Close()
+			for {
+				reader := bufio.NewReader(fpipe)
+				_, err = io.Copy(os.Stdout, reader)
+				if err != nil {
+					logrus.Errorf("error copying data to /dev/stdout, %v", err)
+					break
+				}
+				time.Sleep(100 * time.Millisecond)
```

Comment:
This shouldn’t be necessary. Instead, open a write-only file descriptor before creating this goroutine (then reads don’t fail with EOF; multiple write-only file descriptors for a single pipe are OK, as long as they don’t write simultaneously), and close that file descriptor only when the export is done to trigger EOF in the goroutine.